### PR TITLE
Support named windows in `OVER (window_definition)` clause

### PR DESCRIPTION
**Pull request by Nikita-str**
Issue #999

Parse `window_name` in `OVER ([window_name] [PARTITION BY ..] [ORDER BY ..] ..)`

### CODE REVIEW

File: tests/sqlparser_common.rs [hunk 1]
```diff
@@ -4115,6 +4117,62 @@ fn parse_window_functions() {
         }),
         expr_from_projection(&select.projection[0])
     );
+
+    for i in 0..7 {
```
Comment (alamb) [inline tests/sqlparser_common.rs:4121]: 👍

**Review by alamb — APPROVED** [review]
Thank you for the contribution @Nikita-str  -- this makes sense to me

**Comment by alamb** [comment]
> @alamb first of all, should I do something with parse_deeply_nested_parens_hits_recursion_limits test? Because locally it's failed with stack overflow even in main branch. All others test result is ok.

I think we may need to update the default limit to have a lower limit (it is right on the edge of stack limit in the debug build, and so any change to the stack size leads to a failure)

Maybe we can do it in a separate PR -- any chance you are willing to do so? If not I will try to do so but it may be several days

**Comment by Nikita-str** [comment]
@alamb do you suggest to change `const DEFAULT_REMAINING_DEPTH: usize = 50;` in `src\parser\mod.rs` to `40` for example?

Maybe it's better to wrap this test code into new thread with bigger stack size and then wait until the thread finished?:
```rust
    const KB: usize = 1024;
    const MB: usize = 1024 * KB;
    std::thread::Builder::new().stack_size(8 * MB).spawn(move || { 
        // prev code
    }).unwrap().join().unwrap();
```

**Comment by alamb** [comment]
> Maybe it's better to wrap this test code into new thread with bigger stack size and then wait until the thread finished?:

I agree that seems like a less problematic approach

**Comment by alamb** [comment]
> @alamb do you suggest to change `const DEFAULT_REMAINING_DEPTH: usize = 50;` in `src\parser\mod.rs` to `40` for example?

Yes, this is basically my suggestion

The rationale is that doing so will prevent stack overflows / panics for debug builds.

Another thing we can do is to see if we can reduce the stack frame size for some of the intermediate function calls (reduce stack frame means make fewer local variables)

**Comment by Nikita-str** [comment]
> Maybe we can do it in a separate PR -- any chance you are willing to do so? 
Ok, I will do it, at least by wrapping in new thread.
If I solve it by wrapping then I would to wrap all tests that assert `ParserError::RecursionLimitExceeded` to prevent future the same errors with stack overflow at least in already existed tests, ok?

**Comment by alamb** [comment]
You are not the first to hit this problem -- I think @universalmind303  saw it in https://github.com/sqlparser-rs/sqlparser-rs/pull/1144

Thus i think we need to do some real work -- I will try and find time later this week to help

**Comment by alamb** [comment]
I merged up from main and now this test is passing 🎉

**Comment by alamb** [comment]
Thanks again @Nikita-str -- this is a very nice contribution

**Pull request merged**